crates/turborepo-lib/src/commands/unlink.rs

```diff
@@ -13,3 +25,67 @@ pub fn unlink(base: &mut CommandBase) -> Result<()> {
 
     Ok(())
 }
+
+fn unlink_spaces(base: &mut CommandBase) -> Result<()> {
+    let result =
+        remove_spaces_from_turbo_json(base).context("could not unlink. Something went wrong")?;
+
+    match result {
+        UnlinkSpacesResult::Unlinked => {
+            println!("{}", base.ui.apply(GREY.apply_to("> Unlinked Spaces")));
+        }
+        UnlinkSpacesResult::NoSpacesFound => {
+            println!(
+                "{}",
+                base.ui.apply(GREY.apply_to("> No Spaces config found"))
+            );
+        }
+    }
+
+    Ok(())
+}
+
+pub fn unlink(base: &mut CommandBase, target: Option<LinkTarget>) -> Result<()> {
+    let target = target.unwrap_or(LinkTarget::RemoteCache);
+
+    match target {
+        LinkTarget::RemoteCache => {
+            unlink_remote_caching(base)?;
+        }
+        LinkTarget::Spaces => {
+            unlink_spaces(base)?;
+        }
+    }
+    Ok(())
+}
+
+fn remove_spaces_from_turbo_json(base: &CommandBase) -> Result<UnlinkSpacesResult> {
+    let turbo_json_path = base.repo_root.join("turbo.json");
+
+    if !turbo_json_path.exists() {
+        return Err(anyhow!("turbo.json not found."));
```

Comment:
Should we not add this as context to the error on `File::open`?

Reply:
Hmm - these are two different error types though right? If the file doesn't exist, we can just bail early and not even try and open. 

I did add more error context on the `File::open` call - but if there's a better way to consolidate this let me know!